lib/membrane/aac/parser/config.ex

```diff
@@ -23,10 +24,24 @@ defmodule Membrane.AAC.Parser.Config do
   def parse_config(stream_format) do
     case stream_format.config do
       {:esds, esds} ->
-        Map.merge(stream_format, Esds.parse_esds(esds))
+        esds_stream_format = Esds.parse_esds(esds)
 
-      {:audio_specific_config, config} ->
-        AudioSpecificConfig.parse_audio_specific_config(config)
+        if esds_stream_format.sample_rate != stream_format.sample_rate,
+          do:
+            Membrane.Logger.warning(
+              "Sample rate field decoded from esds differs from stream_format:\nesds:#{inspect(esds_stream_format)}\nstream_format:#{inspect(stream_format)}"
+            )
```

Comment:
You can use multiline string here instead:
```suggestion
            Membrane.Logger.warning("""
            Sample rate field decoded from esds differs from stream_format:
            esds: #{inspect(esds_stream_format)}
            stream_format: #{inspect(stream_format)}
            """)

```